lib/drivers/insuletDriver.js

```diff
@@ -508,6 +508,8 @@ module.exports = function (config) {
       var prog = struct.unpack(basalProgramsHeader.rawdata, prgOffset,
                                fixedRecords.basal_programs_name.format,
                                fixedRecords.basal_programs_name.fields);
+      prog.name = prog.name.replace(/\./g, '-');
```

Comment:
In the main, I agree, but in this particular case I'm not sure what tests are going to gain us. The real thing that would need testing is making sure we're not building *any* objects that have keys that use the illegal characters. Separating out the regex replace functions into a testable module is only going to allow use to test that we can remove illegal characters when we want do; it won't do anything to ensure that we're removing them in all the necessary places.

We should be adding to the jellyfish validation to look for these illegal characters in keys, and at that point we can have ingestion tests. Does that sound like an OK long-term solution?